flink-runtime/src/main/java/org/apache/flink/runtime/operators/sort/CircularQueues.java

```diff
@@ -72,16 +76,23 @@ public void sendResult(MutableObjectIterator<E> result) {
 	}
 
 	@Override
-	public CircularElement<E> take(StageRunner.SortStage stage) {
-		try {
-			return getQueue(stage).take();
-		} catch (InterruptedException e) {
-			throw new WrappingRuntimeException(e);
+	public CircularElement<E> take(StageRunner.SortStage stage) throws InterruptedException {
+		while (!isFinished) {
+			CircularElement<E> value = getQueue(stage).poll(1, TimeUnit.SECONDS);
+			if (value != null) {
+				return value;
+			}
 		}
+		throw new FlinkRuntimeException("The sorter is closed already");
 	}
 
 	@Override
 	public CircularElement<E> poll(StageRunner.SortStage stage) {
 		return getQueue(stage).poll();
 	}
+
+	@Override
+	public void close() throws Exception {
```

Comment:
Sorry but I did not find where we call the `close` method?

Reply:
I am so sorry about that. I completely forgot about calling this method in the `ExternalSorter#close`. I added it there.